tools/mtouch/Application.cs

```diff
@@ -596,6 +874,163 @@ public void SetDefaultFramework ()
 			}
 		}
 
+		void DetectCodeSharing ()
+		{
+			if (AppExtensions.Count == 0)
+				return;
+
+			if (!IsDeviceBuild)
+				return;
+
+			if (NoDevCodeShare) {
+				Driver.Log (2, "Native code sharing has been disabled in the main app, so no code sharing with extensions will occur.");
+				return;
+			}
+
+			if (DeploymentTarget.Major < 8) {
```

Comment:
They could, I was just trying to not fix everything at once, since it makes things a bit more complex.